### ISSUE TITLE
Precise Pretty Printing of durations

### ISSUE THREAD

**Issue by tilsche** [metricq/metricq-python#27]
Print durations, e.g. for db interval configuration, in a human-readable form that can still be parsed again by [`Timedelta.from_string`](https://github.com/metricq/metricq-python/blob/master/metricq/types.py#L49) without loss of accuracy.

That means:
```
Timedelta(1000000000).precise_string == "1s"
Timedelta(100000000).precise_string == "100ms"
Timedelta(1000000001).precise_string == "1000000001ns"
```
- [ ] add more tests

**Comment by tilsche** [comment]
For all Timedeltas `t` this should be true:
`Timedelta.from_string(t.precise_str) == t`

**Comment by phijor** [comment]
Closed by #28.
